velox/functions/prestosql/aggregates/CovarianceAggregates.cpp

```diff
@@ -279,6 +286,97 @@ class CorrIntermediateResult : public CovarIntermediateResult {
   double* m2Y_;
 };
 
+struct RegrAccumulator : public CovarAccumulator {
+  double m2X() const {
+    return m2X_;
+  }
+
+  void update(double x, double y) {
+    double oldMeanX = meanX();
+    CovarAccumulator::update(x, y);
+
+    m2X_ += (x - oldMeanX) * (x - meanX());
```

Comment:
Should we throw exception if `x == meanX()`?  What is the behavior in Presto Java?

Reply:
Presto Java not `check x == meanX()`. https://github.com/prestodb/presto/blob/1edcf2dc2a9ed67ef7a3ce8962601c8246206294/presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationUtils.java#L149

Reply:
Oh never mind it should not matter in this case